components/EmailCollectorForm.vue

```diff
@@ -0,0 +1,185 @@
+<script setup lang="ts">
+import { ref, onMounted } from 'vue'
+
+const props = defineProps({
+  showNoThanks: {
+    type: Boolean,
+    default: false,
+  },
+  isSubmiting: {
+    type: Boolean,
+    default: false,
+  },
+  altDesign: {
+    type: Boolean,
+    default: false,
+  },
+  submissionStatus: {
+    type: String,
+    default: null,
+  },
+  submitButtonText: {
+    type: String,
+    default: 'Sign up',
+  },
+  submitButtonIcon: {
+    type: String,
+    default: null,
+  },
+  thanksMessage: {
+    type: String,
+    default: 'Thank you for signing up!',
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'submitFormEmit', value: any): void
+  (e: 'noThanksEventEmit'): void
+}>()
```

Comment:
ok... I understand. Cool. I guess there is no need to mention it is an emit... because it's clearly an emit... and as a prop, it's preceded with a "@" which clear indicates its is an emit.